### PR TITLE
Endpoints return only base order information

### PR DESCRIPTION
**Pull request by lukaslew**
There is no need to return full order dto object for endpoints like create, batch create or search. The full order dto object is only return by **/order/{id}** endpoint. This should speed up endpoint execution.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -5,6 +5,11 @@ New functionality added in a backwards-compatible manner:
 * [OLMIS-2551](https://openlmis.atlassian.net/browse/OLMIS-2551): Added an endpoint for batch
 order creation, which allows creating multiple orders in one transaction.
 * [OLMIS-2611](https://openlmis.atlassian.net/browse/OLMIS-2611): Added using locale from env file.
+* Order endpoints (except for */orders/{id}*) will now return new, smaller Dto object, which only
```
Comment (palbecki) [inline CHANGELOG.md]: is it "New functionality added in a backwards-compatible manner"?

Comment (lukaslew) [inline CHANGELOG.md]: I updated the file

**Comment by palbecki** [comment]
Should we really return basic dto for POST and get all?

**Comment by lukaslew** [comment]
@palbecki  In my opinion the full order dto object should be return only when this is necessary so when a user try to retrieve it by **/orders/{id}** endpoint. Other endpoints should (in my opinion) return only basic information thanks to the user knows that orders have been created.

**Comment by palbecki** [comment]
@lukaslew I can agree to that but this somehow is breaking our style guide for REST. 
https://github.com/OpenLMIS/openlmis-template-service/blob/master/STYLE-GUIDE.md
PUT, POST: "To prevent an API consumer from having to hit the API again for an updated representation, have the API return the updated (or created) representation as part of the response."
@pgesek What do you think about it? If we merge this our API will be inconsistent.

**Comment by pgesek** [comment]
I would rather leave the full representation for PUT/POST - this shouldn't really impact performance that much anyway, since the call is made from the requisition service, not the client.

**Comment by lukaslew** [comment]
The one problem is that the process of creating a full order dto object takes too much time. The second problem is that the batch create endpoint is used only by the requisition service but it does not handle the response from the fulfillment service. In other words the response is forgotten/abandoned.

File: src/main/resources/api-definition.yaml [hunk 2]
```diff
@@ -10,6 +10,13 @@ documentation:
       content: Welcome to the OpenLMIS fulfillment-service documentation.
 
 schemas:
+  - basicOrderDto: !include schemas/basicOrderDto.json
+  - basicOrderDtoPage: !include schemas/orderDtoPage.json
```
Comment (joshzamor) [inline src/main/resources/api-definition.yaml]: Why is basicOrderDtoPage the same as orderDtoPage?

**Comment by joshzamor** [comment]
I'm concerned about what I think I see happening here.  If I'm reading this right, GET /api/orders returns a different representation than GET /api/orders/{id} ?

**Comment by brandonbowersox** [comment]
In other areas, I think we ended up with /api/thing/{id} being the full representation, and /api/thing/search returning a shorter/basic representation (a list of UUIDs). Is that a true statement? And is that a pattern we could use here with Orders? Specifically, we could create an /api/orders/search endpoint that would return the shorter/basic representation.

File: src/main/resources/api-definition.yaml (hunk 2 above)
Comment (lukaslew) [inline src/main/resources/api-definition.yaml]: this is a typo: I create a new PR with fix for that: https://github.com/OpenLMIS/openlmis-fulfillment/pull/9

**Comment by lukaslew** [comment]
@joshzamor Yes currently both endpoints return different representation of orders. I think we could remove GET /api/orders endpoint because it returns all orders from database and there is no check if a user has permission for that. Also I don't think this endpoint is used by the UI.

@brandonbowersox Currently the fulfillment service contains the search endpoint and it returns the shorter/basic representation of orders because the full representation was not used and some data was return with orders (like order line items) but they were not used by UI. The GET /api/orders/{id} returns full representation.

**Comment by pgesek** [comment]
btw, GitHub does not send notifications for closed prs apparently, so I've completely missed these comments

**Comment by brandonbowersox** [comment]
@lukaslew It sounds like as a follow-up step, we should come back and re-consider /api/orders endpoint. Based on what you say (if it returns all data with no checks), we need to address that. I think we need to decide if that endpoint should return full representations as well. And if the UI does not need or use that anyway, that's part of the consideration.

@joshzamor I would be interested to know how you would see us addressing it. I can create a stub ticket, but let me know if you have specific ideas.

@pgesek good to know!

**Pull request merged**